CI: SetUpTest hitting runtime error: invalid memory address or nil pointer dereference

Seen in v1.1 branch:
```
[...]
09:42:59 level=info msg="Initializing identity allocator"
09:42:59 level=info msg="Starting IP identity watcher" subsys=ipcache
09:42:59 level=info msg="Opened access log" file-path=/tmp/cilium-test-run914382316/cilium-access.log
09:42:59 level=info msg="Envoy: Starting xDS gRPC server listening on /tmp/cilium-test-run914382316/xds.sock"
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 PANIC: daemon_test.go:158: DaemonEtcdSuite.SetUpTest
09:42:59 
09:42:59 ... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0xF3B38A)
09:42:59 
09:42:59 /usr/local/go/src/runtime/panic.go:502
09:42:59   in gopanic
09:42:59 /usr/local/go/src/runtime/panic.go:63
09:42:59   in panicmem
09:42:59 /usr/local/go/src/runtime/signal_unix.go:388
09:42:59   in sigpanic
09:42:59 state.go:155
09:42:59   in Daemon.regenerateRestoredEndpoints
09:42:59 daemon.go:1291
09:42:59   in NewDaemon
09:42:59 daemon_test.go:105
09:42:59   in DaemonSuite.SetUpTest
09:42:59 daemon_test.go:160
09:42:59   in DaemonEtcdSuite.SetUpTest
09:42:59 /usr/local/go/src/reflect/value.go:308
09:42:59   in Value.Call
09:42:59 /usr/local/go/src/runtime/asm_amd64.s:2361
09:42:59   in goexit
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 PANIC: <autogenerated>:1: DaemonEtcdSuite.TestEndpointAddInvalidLabel
09:42:59 
09:42:59 ... Panic: Fixture has panicked (see related PANIC)
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestEndpointAddNoLabels
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestEndpointAddReservedLabel
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestGetMaxServiceID
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestK8sErrorLogTimeout
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestMinimumWorkerThreadsIsSet
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestParseKernelVersion
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestReadEPsFromDirNames
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestRemovePolicy
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestReplacePolicy
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestServices
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestSyncLabels
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestUpdateConsumerMap
09:42:59 MISS: <autogenerated>:1: DaemonEtcdSuite.TestUpdateSecLabels
09:42:59 level=info msg="Waiting for consul to elect a leader"
09:42:59 level=info msg="Valid label prefix configuration:"
09:42:59 level=info msg=" - :io.kubernetes.pod.namespace"
09:42:59 level=info msg=" - :io.cilium.k8s.namespace.labels"
09:42:59 level=info msg=" - !:io.kubernetes"
09:42:59 level=info msg=" - !:.*kubernetes.io"
09:42:59 level=info msg=" - !:pod-template-generation"
09:42:59 level=info msg=" - !:pod-template-hash"
09:42:59 level=info msg=" - !:controller-revision-hash"
09:42:59 level=info msg=" - !:annotation.cilium.io/"
09:42:59 level=info msg=" - !:annotation.cilium-identity"
09:42:59 level=info msg=" - !:annotation.sidecar.istio.io"
09:42:59 level=info msg="Clearing leftover Cilium veths"
09:42:59 level=info msg="Launching endpoint builder workers"
09:42:59 level=info msg="Initializing node addressing"
09:42:59 level=info msg="Initializing IPAM"
09:42:59 level=info msg="Restoring endpoints from former life..."
09:42:59 level=error msg="Unable to restore existing endpoints" error="unable to read BPF endpoint list: Unable to get object /sys/fs/bpf/tc/globals/cilium_lxc: no such file or directory"
09:42:59 level=info msg="Validating configured node address ranges"
09:42:59 level=info msg="Addressing information:"
09:42:59 level=info msg="  Local node-name: 57c19a7cce59"
09:42:59 level=info msg="  Node-IPv6: fc00::10ca:1"
09:42:59 level=info msg="  External-Node IPv4: 172.19.0.4"
09:42:59 level=info msg="  Internal-Node IPv4: 10.4.0.1"
09:42:59 level=info msg="  Cluster IPv6 prefix: f00d::/64"
09:42:59 level=info msg="  Cluster IPv4 prefix: 10.0.0.0/8"
09:42:59 level=info msg="  IPv6 node prefix: f00d::a04:0:0:0/96"
09:42:59 level=info msg="  IPv6 allocation prefix: f00d::a04:0:0:0/112"
09:42:59 level=info msg="  IPv4 allocation prefix: 10.4.0.0/16"
09:42:59 level=info msg="  IPv6 router address: f00d::a04:0:0:1"
09:42:59 level=info msg="  Loopback IPv4: 10.4.89.154"
09:42:59 level=info msg="Adding local node to local cluster node list"
09:42:59 level=error msg="bpf: Unable to update in tunnel endpoint map" error="Unable to get object /sys/fs/bpf/tc/globals/cilium_tunnel_map: no such file or directory" ipAddr=10.4.0.0/16
09:42:59 level=error msg="bpf: Unable to update in tunnel endpoint map" error="Unable to get object /sys/fs/bpf/tc/globals/cilium_tunnel_map: no such file or directory" ipAddr="f00d::a04:0:0:0/112"
09:42:59 level=info msg="Opened access log" file-path=/tmp/cilium-test-run382793307/cilium-access.log
09:42:59 level=info msg="Envoy: Starting xDS gRPC server listening on /tmp/cilium-test-run382793307/xds.sock"
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 PANIC: daemon_test.go:173: DaemonConsulSuite.SetUpTest
09:42:59 
09:42:59 ... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0xF3B38A)
09:42:59 
09:42:59 /usr/local/go/src/runtime/panic.go:502
09:42:59   in gopanic
09:42:59 /usr/local/go/src/runtime/panic.go:63
09:42:59   in panicmem
09:42:59 /usr/local/go/src/runtime/signal_unix.go:388
09:42:59   in sigpanic
09:42:59 state.go:155
09:42:59   in Daemon.regenerateRestoredEndpoints
09:42:59 daemon.go:1291
09:42:59   in NewDaemon
09:42:59 daemon_test.go:105
09:42:59   in DaemonSuite.SetUpTest
09:42:59 daemon_test.go:175
09:42:59   in DaemonConsulSuite.SetUpTest
09:42:59 /usr/local/go/src/reflect/value.go:308
09:42:59   in Value.Call
09:42:59 /usr/local/go/src/runtime/asm_amd64.s:2361
09:42:59   in goexit
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 PANIC: <autogenerated>:1: DaemonConsulSuite.TestEndpointAddInvalidLabel
09:42:59 
09:42:59 ... Panic: Fixture has panicked (see related PANIC)
09:42:59 
09:42:59 ----------------------------------------------------------------------
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestEndpointAddNoLabels
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestEndpointAddReservedLabel
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestGetMaxServiceID
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestK8sErrorLogTimeout
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestMinimumWorkerThreadsIsSet
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestParseKernelVersion
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestReadEPsFromDirNames
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestRemovePolicy
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestReplacePolicy
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestServices
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestSyncLabels
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestUpdateConsumerMap
09:42:59 MISS: <autogenerated>:1: DaemonConsulSuite.TestUpdateSecLabels
09:42:59 OOPS: 0 passed, 2 FIXTURE-PANICKED, 28 MISSED
09:42:59 --- FAIL: Test (60.19s)
```

--- a/daemon/state.go
+++ b/daemon/state.go
@@ -67,7 +67,7 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 
 	existingEndpoints, err := lxcmap.DumpToMap()
 	if err != nil {
-		return nil, err
+		return state, err
 	}
 
 	dirFiles, err := ioutil.ReadDir(dir)

